### PR TITLE
FAT-3033/C347862

### PR DESCRIPTION
**Pull request by re-jas**
https://foliotest.testrail.io/index.php?/cases/view/347862

### CODE REVIEW

File: cypress/e2e/recieve-notice/recieve-notice-triggers.cy.js [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+        testData.ruleProps.l = loanPolicyId;
+        testData.ruleProps.l = CirculationRules.addRuleViaApi(
+          testData.baseRules,
+          testData.ruleProps,
+          'g ',
+          patronGroup.id
+        );
+      });
+
+      const checkMail = (checkParams) => {
```
Comment (penfriend) [inline cypress/e2e/recieve-notice/recieve-notice-triggers.cy.js:277]: it's better to rename to checkNoticeIsSent

File: cypress/e2e/recieve-notice/recieve-notice-triggers.cy.js [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+        cy.visit(TopMenu.circulationLogPath);
+        SearchPane.searchByUserBarcode(userData.barcode);
+        SearchPane.checkResultSearch(checkParams);
+      };
```
Comment (penfriend) [inline cypress/e2e/recieve-notice/recieve-notice-triggers.cy.js:281]: Please this method outside the "it" block and before the "before" block

File: cypress/e2e/recieve-notice/recieve-notice-triggers.cy.js [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+      const createPatronNoticeTemplate = (template) => {
+        NewNoticePolicyTemplate.startAdding();
+        NewNoticePolicyTemplate.checkInitialState();
+        NewNoticePolicyTemplate.addToken('item.title');
+        NewNoticePolicyTemplate.create(template, false);
+        NewNoticePolicyTemplate.checkPreview();
+        NewNoticePolicyTemplate.saveAndClose();
+        NewNoticePolicyTemplate.waitLoading();
+        NewNoticePolicyTemplate.checkAfterSaving(template);
+      };
```
Comment (penfriend) [inline cypress/e2e/recieve-notice/recieve-notice-triggers.cy.js:249]: Please this method outside the "it" block and before the "before" block

File: cypress/e2e/recieve-notice/recieve-notice-triggers.cy.js [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+      NewNoticePolicy.addNotice(noticePolicy.selectOptions(loanDueDateChangeTemplate), 1);
+      NewNoticePolicy.addNotice(noticePolicy.selectOptions(checkInTemplate), 2);
+      NewNoticePolicy.save();
+      NewNoticePolicy.waitLoading();
+      NewNoticePolicy.check(noticePolicy);
+
+      cy.getNoticePolicy({ query: `name=="${noticePolicy.name}"` }).then((res) => {
+        testData.ruleProps.n = res[0].id;
+        testData.ruleProps.l = loanPolicyId;
+        testData.ruleProps.l = CirculationRules.addRuleViaApi(
```
Comment (penfriend) [inline cypress/e2e/recieve-notice/recieve-notice-triggers.cy.js:269]: Does method CirculationRules.addRuleViaApi return somethig? Do you need it to save the result of this method to the property testData.ruleProps.l?

File: cypress/support/fragments/loans/loansPage.js [hunk 5]
```diff
@@ -14,6 +17,9 @@ export default {
       Button('Change due date').click(),
     ]);
   },
+  openChangeDueDate() {
+    cy.do(Button('Change due date').click());
```
Comment (penfriend) [inline cypress/support/fragments/loans/loansPage.js:21]: move to const Button('Change due date')

**Review by MaksimTerekhov — APPROVED** [review]

**Review by penfriend — APPROVED** [review]

**Pull request merged**